### PR TITLE
SONARIAC-508 Rule S6364: Exclude Aurora for Cloudformation

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/checks/PropertyUtils.java [hunk 1]
```diff
@@ -48,6 +49,13 @@ public static Trilean has(@Nullable Tree tree, String key) {
     return Trilean.FALSE;
   }
 
+  public static boolean hasKeyWithValue(@Nullable Tree tree, String key, Collection<String> values) {
```
Comment (nils-werner-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/checks/PropertyUtils.java:52]: IMO this method is too specific and the name is not self-explaining. Due to we already have a helper value, I could imagine a helper like `valueIs(@Nullable Tree tree, String key, Predicate<Tree>)` WDYT?

File: iac-extensions/cloudformation/src/main/java/org/sonar/iac/cloudformation/checks/ShortBackupRetentionCheck.java [hunk 2]
```diff
@@ -41,7 +43,9 @@ public class ShortBackupRetentionCheck extends AbstractResourceCheck {
 
   @Override
   protected void checkResource(CheckContext ctx, Resource resource) {
-    if ((resource.isType("AWS::RDS::DBInstance") && PropertyUtils.isMissing(resource.properties(), "SourceDBInstanceIdentifier"))
+    if ((resource.isType("AWS::RDS::DBInstance")
+        && PropertyUtils.isMissing(resource.properties(), "SourceDBInstanceIdentifier")
+        && !PropertyUtils.hasKeyWithValue(resource.properties(), "Engine", ENGINES_EXCEPTION))
```
Comment (nils-werner-sonarsource) [inline iac-extensions/cloudformation/src/main/java/org/sonar/iac/cloudformation/checks/ShortBackupRetentionCheck.java:48]: I would switch the conditions.

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM but see my comments about the helper method

File: iac-common/src/main/java/org/sonar/iac/common/checks/PropertyUtils.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-common/src/main/java/org/sonar/iac/common/checks/PropertyUtils.java:52]: You are right, I will change this method to have a less specific one.

**Pull request merged**